### PR TITLE
Make unit tests compatible with Python 2.7.5

### LINKED ISSUE #116: Issue on python 2.7.5 running unittests
When running unittests, on python  2.7.5, I get some failures.

I'm not running virtualenv

<pre>
[myork@z820 nailgun]$ cat /etc/redhat-release 
Red Hat Enterprise Linux Workstation release 7.1 (Maipo)
</pre>

<pre>
>>> print sys.version
2.7.5 (default, Feb 11 2014, 07:46:25) 
[GCC 4.8.2 20140120 (Red Hat 4.8.2-13)]
</pre>

Unittests are all run from within pycharm.

<pre>
/usr/bin/python2.7 /home/myork/applications/pycharm-4.5.1/helpers/pycharm/utrunner.py /home/myork/PycharmProjects/nailgun/ true
Testing started at 15:02 ...

Failure
Expected :"test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'))"
Actual   :"tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'))"
 <Click to see difference>

Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_entity_mixins.py", line 311, in test_repr_v1
    self.assertEqual(repr(entity), target)
AssertionError: "test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'))" != "tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'))"

Failure
Expected :"test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), id=2495229732896996198)"
Actual   :"tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), id=2495229732896996198)"
 <Click to see difference>

Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_entity_mixins.py", line 329, in test_repr_v2
    self.assertEqual(repr(entity), target)
AssertionError: "test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), id=2495229732896996198)" != "tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), id=2495229732896996198)"

Failure
Expected :"test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), one_to_many=[test_entity_mixins.SampleEntity(nailgun.config.ServerConfig(url='http://example.com'), id=3965265452630526438)])"
Actual   :"tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), one_to_many=[tests.test_entity_mixins.SampleEntity(nailgun.config.ServerConfig(url='http://example.com'), id=3965265452630526438)])"
 <Click to see difference>

Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_entity_mixins.py", line 354, in test_repr_v3
    self.assertEqual(repr(entity), target)
AssertionError: "test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), one_to_many=[test_entity_mixins.SampleEntity(nailgun.config.ServerConfig(url='http://example.com'), id=3965265452630526438)])" != "tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), one_to_many=[tests.test_entity_mixins.SampleEntity(nailgun.config.ServerConfig(url='http://example.com'), id=3965265452630526438)])"

Failure
Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 74, in test_get
    _compare_configs(self, config, server_config)
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 44, in _compare_configs
    self.assertEqual(dict_config, cfg)
AssertionError: {'url': 'booger', 'version': '1.2.3.4.dev5+67'} != {'url': u'booger', 'version': "(u'00000001', u'00000002', u'00000003', u'0000000 [truncated]...
- {'url': 'booger', 'version': '1.2.3.4.dev5+67'}
+ {'url': u'booger',
+  'version': "(u'00000001', u'00000002', u'00000003', u'00000004', '*@', u'00000005', u'*+', u'00000067', '*final')"}

Failure
Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 58, in test_init
    _compare_configs(self, config, server_config)
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 44, in _compare_configs
    self.assertEqual(dict_config, cfg)
AssertionError: {'url': 'booger', 'version': '1.2.3.4.dev5+67'} != {'url': 'booger', 'version': "('00000001', '00000002', '00000003', '00000004', ' [truncated]...
- {'url': 'booger', 'version': '1.2.3.4.dev5+67'}
+ {'url': 'booger',
+  'version': "('00000001', '00000002', '00000003', '00000004', '*@', '00000005', '*+', '00000067', '*final')"}

Failure
Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 150, in test_init
    _compare_configs(self, config, ServerConfig(**config))
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 44, in _compare_configs
    self.assertEqual(dict_config, cfg)
AssertionError: {'url': 'booger', 'version': '1.2.3.4.dev5+67'} != {'url': 'booger', 'version': "('00000001', '00000002', '00000003', '00000004', ' [truncated]...
- {'url': 'booger', 'version': '1.2.3.4.dev5+67'}
+ {'url': 'booger',
+  'version': "('00000001', '00000002', '00000003', '00000004', '*@', '00000005', '*+', '00000067', '*final')"}

Process finished with exit code 0
</pre>

### PR DESCRIPTION
**Pull request by Ichimonji10**
Fix #116.

### CODE REVIEW

**Comment by Ichimonji10** [comment]
@my0373 please checkout this branch and let me know if it fixes the issues you're encountering.

**Comment by my0373** [comment]
ACK

Worked perfectly
<pre>
[myork@z820 nailgun]$ make test
python -m unittest discover --start-directory tests --top-level-directory .
.........................................................................................................................................
----------------------------------------------------------------------
Ran 137 tests in 0.215s

OK
</pre>

**Comment by Ichimonji10** [comment]
:tada:

**Comment by Ichimonji10** [comment]
I'd guess that the pycharm unittest runner `cd`'s to the `tests` directory before running tests.

**Comment by my0373** [comment]
I think you are right, lesson learned. Trust <pre>make test</pre> and no-one else :)

File: tests/test_config.py [hunk 1]
```diff
@@ -38,10 +38,10 @@ def _compare_configs(self, dict_config, server_config):
         subclass thereof.
 
     """
-    cfg = vars(server_config).copy()
-    if 'version' in cfg:
-        cfg['version'] = str(cfg['version'])
-    self.assertEqual(dict_config, cfg)
+    if 'version' in dict_config:
+        dict_config = dict_config.copy()  # shadow the passed in dict
```
Comment (omaciel) [inline tests/test_config.py]: Shadow the <blank> passed?

Comment (Ichimonji10) [inline tests/test_config.py]: What I'm doing here is making a new variable called `dict_config` that is completely separate from the passed-in `dict_config`. Maybe this example will explain things a bit better:

```python
>>> # This is baaaaad.
>>> def do_not_shadow(arg):
...     arg[1] = 5
...
>>> arg = {1: 2}
>>> arg
{1: 2}
>>> do_not_shadow(arg)
>>> arg
{1: 5}
>>> # This is good!
>>> def do_shadow(arg):
...     arg = arg.copy()
...     arg[1] = 5
...
>>> arg = {1: 2}
>>> do_shadow(arg)
>>> arg
{1: 2}
```

**Comment by elyezer** [comment]
ACK

**Comment by Ichimonji10** [comment]
Rebased.

File: tests/test_config.py (hunk 1 above)
Comment (omaciel) [inline tests/test_config.py]: Sorry, I misread the docstring and thought that you were missing a word for the 'what' :) ACK

Comment (Ichimonji10) [inline tests/test_config.py]: Ahhh, gotcha. :smile:

**Pull request merged**